src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx

```diff
@@ -36,111 +40,375 @@ describe('ComboBox', () => {
+      simulateInputChange(input, searchTerm);
+      const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.length(21);
+
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+      fireEvent.keyDown(input, { key: 'Tab' });
+
+      expect(input).to.have.value(searchTerm);
+      expect(listbox).to.have.length(0);
+    });
+
+    it('should close the list and retain input on Escape', () => {
+      const searchTerm = 'm';
+      const { getByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.length(21);
+
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+      fireEvent.keyDown(input, { key: 'Escape' });
+
+      expect(input).to.have.value(searchTerm);
+      expect(listbox).to.have.length(0);
+    });
+  });
+
+  describe('Accessibility', () => {
+    it('should have combobox with message-aria-describedby', () => {
+      const { getByTestId } = render(<ComboBox {...props} />);
+
+      const combobox = getByTestId('combobox-input');
+
+      expect(combobox).to.have.attribute(
+        'message-aria-describedby',
+        `\n      When autocomplete results are available use up and down arrows to\n      review and enter to select. Touch device users, explore by touch or\n      with swipe gestures.\n    `,
+      );
+    });
+
+    it('should have listbox with role and aria-label', () => {
+      const { getByRole } = render(<ComboBox {...props} />);
+
+      const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.attribute('role', 'listbox');
+      expect(listbox).to.have.attribute(
+        'aria-label',
+        'List of matching conditions',
+      );
+    });
+
+    it('should apply aria-activedescendant correctly when navigating the list', () => {
+      const searchTerm = 'u';
+      const { getByTestId, getByRole } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.attribute('aria-activedescendant', 'option-0');
+    });
+
+    it('should provide correct screen reader feedback (aria-live regions)', () => {
+      const searchTerm = 's';
+      const searchResults = fullStringSimilaritySearch(searchTerm, items);
+      const freeTextAndFilteredItemsCount = searchResults.length + 1;
+      const { getByTestId, getByText } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const screenReaderMessage = getByText(
+        `${freeTextAndFilteredItemsCount} results available.`,
+      );
+
+      expect(screenReaderMessage).to.exist;
+      expect(screenReaderMessage).to.have.attribute('role', 'alert');
```

Comment:
Just FYI as we talked about earlier I was considering removing this `role="alert"` because I thought it might be overwritten by the explicit setting of `aria-live="polite"` and `aria-atomic`.  I need to do more testing tho. It might just stay in.